apps/backend/src/libs/types/repository.type.ts

```diff
@@ -1,8 +1,10 @@
+import { type PaginationQueryParameters } from "~/libs/types/types.js";
+
 type Repository<T = unknown> = {
 	create(payload: unknown): Promise<T>;
 	delete(id: number): Promise<boolean>;
 	find(id: number): Promise<null | T>;
-	findAll(): Promise<T[]>;
+	findAll(parameters: PaginationQueryParameters): Promise<{ items: T[] }>;
```

Comment:
let's make it unknown for repository
```suggestion
	findAll(payload: unknown): Promise<{ items: T[] }>;
```